src/node/txreconciliation.h

```diff
@@ -122,6 +122,16 @@ class TxReconciliationTracker
      * Check if a peer is registered to reconcile transactions with us.
      */
     bool IsPeerRegistered(NodeId peer_id) const;
+
+
+    /**
+     * Returns a collections of node ids sorted by how many instances of the provided transaction ids
+     * can be found in their reconciliation sets (from less to more). If two values tie, they will be
+     * sorted in order or addition, which depends on the internal order of [m_states].
+     * This is meant to be called with a collection of ancestor ids for a given transaction in order to
+     * decide whether the transaction should be fanout with ancestors or not.
+    */
+    std::vector<NodeId> SortPeersByFewestParents(std::vector<Wtxid> parents);
```

Comment:
I think the rationale for introducing `SortPeersByFewestParents` could be more fleshed out, i.e iiuc explaining that eligible peers with the lowest number of parents for a target transaction in their reconciliation sets are favored for fan-out announcements, as this can be a hint of better connectivity in the overall transaction-relay graph and we wish to propagate the target transaction faster (..?).

Reply:
The rationale for who this method is used is provided in `RelayTransaction` when the method is used

https://github.com/bitcoin/bitcoin/blob/804ccf030362fe5b232fe5c4213b77899a425f2b/src/net_processing.cpp#L2154-L2159

I think the docs give enough rationale as to what this is intended for without being excessively narrow as to what the use cases could be.